js/__internal/viz/chart_components/m_advanced_chart.ts

```diff
@@ -0,0 +1,863 @@
+
+  return Math.min(width, height) * maxSize;
+}
+
+function setAxisVisualRangeByOption(arg, axis, isDirectOption, index?) {
+  let options;
+  let visualRange;
+
+  if (isDirectOption) {
+    visualRange = arg.value;
+    options = {
+      skipEventRising: true,
+    };
+
+    const wrappedVisualRange = wrapVisualRange(arg.fullName, visualRange);
+    if (wrappedVisualRange) {
+      options = { allowPartialUpdate: true };
+      visualRange = wrappedVisualRange;
+    }
+  } else {
+    visualRange = (_isDefined(index) ? arg.value[index] : arg.value).visualRange;
+  }
+  axis.visualRange(visualRange, options);
+}
+
+function getAxisTypes(groupsData, axis, isArgumentAxes) {
+  if (isArgumentAxes) {
+    return { argumentAxisType: groupsData.argumentAxisType, argumentType: groupsData.argumentType };
+  }
+  const { valueAxisType, valueType } = groupsData.groups.filter((g) => g.valueAxis === axis)[0];
+  return { valueAxisType, valueType };
+}
+
+function wrapVisualRange(fullName, value) {
+  const pathElements = fullName.split('.');
+  const destElem = pathElements[pathElements.length - 1];
+
+  if (destElem === 'endValue' || destElem === 'startValue') {
+    return {
+      [destElem]: value,
+    };
+  }
+  return undefined;
+}
+
+export const AdvancedChart = BaseChart.inherit({
+
+  _fontFields: [`${COMMON_AXIS_SETTINGS}.label.${FONT}`, `${COMMON_AXIS_SETTINGS}.title.${FONT}`],
+
+  _partialOptionChangesMap: {
+    visualRange: VISUAL_RANGE,
+    _customVisualRange: VISUAL_RANGE,
+    strips: 'REFRESH_AXES',
+    constantLines: 'REFRESH_AXES',
+  },
+
+  _partialOptionChangesPath: {
+    argumentAxis: ['strips', 'constantLines', 'visualRange', '_customVisualRange'],
+    valueAxis: ['strips', 'constantLines', 'visualRange', '_customVisualRange'],
+  },
+
+  _initCore() {
+    this._panesClipRects = {};
+    this.callBase();
+  },
+
+  _disposeCore() {
+    const disposeObjectsInArray = this._disposeObjectsInArray;
+    const panesClipRects = this._panesClipRects;
+
+    this.callBase();
+    disposeObjectsInArray.call(panesClipRects, 'fixed');
+    disposeObjectsInArray.call(panesClipRects, 'base');
+    disposeObjectsInArray.call(panesClipRects, 'wide');
+    this._panesClipRects = null;
+    this._labelsAxesGroup.linkOff();
+    this._labelsAxesGroup.dispose();
+    this._labelsAxesGroup = null;
+  },
+
+  _dispose() {
+    const that = this;
+    const disposeObjectsInArray = this._disposeObjectsInArray;
+
+    that.callBase();
+
+    that.panes = null;
+    if (that._legend) {
+      that._legend.dispose();
+      that._legend = null;
+    }
+    disposeObjectsInArray.call(that, 'panesBackground');
+    disposeObjectsInArray.call(that, 'seriesFamilies');
+    that._disposeAxes();
+  },
+
+  _createPanes() {
+    this._cleanPanesClipRects('fixed');
+    this._cleanPanesClipRects('base');
+    this._cleanPanesClipRects('wide');
+  },
+
+  _cleanPanesClipRects(clipArrayName) {
+    const clipArray = this._panesClipRects[clipArrayName];
```

Comment:
It's better to move default init here
```suggestion
    const clipArray = this._panesClipRects[clipArrayName] ?? [];
```